src/Middleware/Csrf.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Middleware;
+
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Security\Random;
+use Yiisoft\Security\TokenMasker;
+use Yiisoft\Yii\Web\Cookie;
+use Yiisoft\Yii\Web\Session\SessionInterface;
+
+final class Csrf implements MiddlewareInterface
+{
+    public const CSRF_HEADER = 'X-CSRF-Token';
+
+    private $name = '_csrf';
+    private $cookieParams = ['httpOnly' => true];
+    private $enableCookie = true;
```

Comment:
It could be removed because we do not support cookie-less sessions so we can assume that cookies are always available.